core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4325,21 +4325,18 @@ int RootClingMain(int argc,
 
    interp.getOptions().ErrorOut = true;
    interp.enableRawInput(true);
-   if (isGenreflex) {
-      if (interp.declare("namespace std {} using namespace std;") != cling::Interpreter::kSuccess) {
-         // There was an error.
-         ROOT::TMetaUtils::Error(0, "Error loading the default header files.\n");
-         return 1;
-      }
-   } else {
-      // rootcling
-      if (interp.declare("namespace std {} using namespace std;") != cling::Interpreter::kSuccess
-            // CINT uses to define a few header implicitly, we need to do it explicitly.
-            || interp.declare("#include <assert.h>\n"
-                             ) != cling::Interpreter::kSuccess
-            || interp.declare("#include \"Rtypes.h\"\n"
-                              "#include \"TObject.h\""
-                             ) != cling::Interpreter::kSuccess
+   if (interp.declare("namespace std {} using namespace std;") != cling::Interpreter::kSuccess) {
+      ROOT::TMetaUtils::Error(0, "Error loading the default header files.\n");
+      return 1;
+   }
+   if (!isGenreflex) { // rootcling
+      // ROOTCINT uses to define a few header implicitly, we need to do it explicitly.
+      if (interp.declare("#include <assert.h>\n"
+                         "#include <stdlib.h>\n"
```

Comment:
You recently removed those.  The commit message does not indicate a reason for their reintroduction.  Is this a branch merge issue or is it intentional?

Reply:
The things we removed were in TCling.

Reply:
Could augment the commit log to add why those are being added here then?

Reply:
They are not added but refactored. The code is as it was before.

Reply:
I somehow can't see the corresponding code in the left/red column, what am I missing?

Reply:
I am on my phone but here is what I did: moved the using clause as it was common for genreflex and rootcling. Then inverted the if-stmt as the genreflex case became a nop. I may have screwed things up but I do not see that.

Reply:
Ah, i see what you mean. I screwed up rebasing...

Reply:
thanks.